src/CMakeLists.txt

```diff
@@ -272,6 +273,10 @@ if(${ENABLE_COVERAGE})
     -fprofile-arcs
     -ftest-coverage
     -O0)
+  add_link_options(
+   -lgcov
+   --coverage
```

Comment:
could you take this opportunity to replace `-fprofile-arcs` and `-ftest-coverage` with `--coverage` in `add_compile_option()` call above? so that it is more consistent and simpler this way, as, IIUC, `--coverage` covers these two options. see https://llvm.org/docs/CommandGuide/llvm-cov.html and https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html#index-gcov . also, i think we can drop `-lgcov` here.

Reply:
Good spot - have changed those and re-tested